src/lightning/data/streaming/writer.py

```diff
@@ -212,39 +210,42 @@ def _serialize(self, item: Any, sizes: List[int], data: List[bytes]) -> str:
 
     def _create_chunk(self, filename: str, on_done: bool = False) -> bytes:
         """Create a binary chunk from all the binarized items."""
+        items = []
+
         if on_done:
             indices = sorted(self._serialized_items.keys())
             for i in range(len(indices) - 1):
                 assert indices[i] == indices[i + 1] - 1, indices
-            min_index = indices[0]
-            max_index = indices[-1] + 1
-            num_items = np.uint32(max_index - min_index)
```

Comment:
Correct me if I'm wrong, but num_items is always len(items) right? So we don't need to compute it this complicated way, and I just moved the definition below.

Reply:
Yes.